cmds/core/sync/sync.go

```diff
@@ -1,12 +1,68 @@
-// Copyright 2016-2017 the u-root Authors. All rights reserved
+// Copyright 2016-2020 the u-root Authors. All rights reserved
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
 // sync command in Go.
+//
+// Synopsis:
+//		sync [-df] [FILE]
+//
+
 package main
 
-import "syscall"
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"syscall"
+)
+
+const SYS_SYNCFS = 306
+
+var (
+	data = flag.Bool("d", false, "sync file data, no metadata")
+	fs   = flag.Bool("f", false, "commit filesystem caches to disk")
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTION] [FILE]...\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+}
 
 func main() {
-	syscall.Sync()
+	flag.Parse()
+	if *data && *fs == true {
+		log.Fatal("sync: cannot specify both --data and --file-system")
+	}
+
+	if flag.NArg() < 1 {
+		syscall.Sync()
```

Comment:
you might add an os.Exit(0) and drop the else ...

Reply:
sync never fails ... but golangcibot may not know that. You might need to blank this out.